consul/data_source_consul_config_entry_v2_exported_services.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package consul
+
+import (
+	"encoding/json"
+	"fmt"
+
+	pbmulticluster "github.com/hashicorp/consul/proto-public/pbmulticluster/v2"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+func dataSourceConsulConfigEntryV2ExportedServices() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceConsulV2ExportedServicesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the config entry to read.",
+			},
+
+			"kind": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The kind of exported services config (ExportedServices, NamespaceExportedServices, PartitionExportedServices).",
+			},
+
+			"partition": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The partition the config entry is associated with.",
+			},
+
+			"namespace": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The namespace the config entry is associated with.",
+			},
+
+			"services": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported services.",
+			},
+
+			"partition_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service partition consumers.",
+			},
+			"peer_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service peer consumers.",
+			},
+			"sameness_group_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service sameness group consumers.",
+			},
+		},
+	}
+}
+
+func dataSourceConsulV2ExportedServicesRead(d *schema.ResourceData, meta interface{}) error {
+	client, qOpts, _ := getClient(d, meta)
+	name := d.Get("name").(string)
+	kind := d.Get("kind").(string)
+	gvk := &GVK{ //&api.GVK{
+		Group:   "multicluster",
+		Version: "v2",
+		Kind:    kind,
+	}
+	resp, err := v2MulticlusterRead(client, gvk, name, qOpts)
+	if err != nil || resp == nil {
+		return fmt.Errorf("exported services config not found: %s", name)
+	}
+	respData, err := json.Marshal(resp["data"])
```

Comment:
I think if the "data" field don't exist (for some reason) this will return an empty `pbmulticluster.ExportedServices` struct, is that the behaviour we want? 🤔
Same for "ID"